freqtrade/strategy/interface.py

```diff
@@ -381,6 +384,27 @@ def custom_stake_amount(self, pair: str, current_time: datetime, current_rate: f
         """
         return proposed_stake
 
+    def adjust_trade_position(self, pair: str, trade: Trade, current_time: datetime,
+                              current_rate: float, current_profit: float, **kwargs
+                              ) -> Optional[float]:
+        """
```

Comment:
Thinking about the discussion we're just having - wouldn't we want to have min/max stake available as parameters here too?

That would help stay within "tradable" limits i think - as you'd know how much is available - as well as how much you MUST buy for the order to go through (10$ - for example).

Reply:
I'm not sure about the buy_order_count - Yes, the example i've provided and what most people are using is relying on the previous buy order count, but only to calculate the next amount via multiplication. 

The strategy I'm personally currently testing is actually using the previously bought pair stake amount sum, meaning the count itself is meaningless, it's the stake amount that has already been invested into trade that matters.

Also, with some orders getting partially filled and some failing, the "count" and actual value of those orders could differ a lot.

Reply:
Regarding "min/max stake available as parameters" - Yes, technically we are providing those to custom_stake_amount so probably makes sense to add them here also.

Reply:
buy_count idea is now also resolved by having trade.nr_of_successful_buys() and trade.nr_of_successful_sells() functions available straight at trade object, allowing us to use them in multiple callbacks without having to add callback arguments and calculating them if they are not actually used in the implementation.